### ISSUE TITLE
RFC: Automatic Refetching for Graphcache

### ISSUE THREAD

**Issue by arturenault** [FormidableLabs/urql#2375]
## Summary
We tried switching from document cache to graphcache, but were surprised to learn that it doesn't support any kind of [automatic refetching](https://formidable.com/open-source/urql/docs/comparison/).

With the document cache, we were able to automatically invalidate several queries running in the background with no additional code. However we ran into some caching issues where` __typenames` weren't available. Graphcache solved those issues, but we found that queries weren't being automatically refetched/revalidated.

I understand that graphcache tries to reduce refetches as much as possible, but it would be great to have a setting that mimics document cache's behavior so we wouldn't have to write hundreds of updaters to be able to use graphcache. 

## Proposed Solution
Add a setting that approximates document cache's behavior with some improvements. Here's an example of how it could work:

1. The client performs a query to get a User. It returns something like `{ id: 1 }`. Graphcache caches the result to its key.
2. Graphcache caches that to its key (e.g. `User:1`)
3. The client performs a mutation to update a user. It also returns something like `{ id: 1}`.
4. Graphcache can now invalidate the initial query and force a refetch.

Basically, if the setting were enabled and we perform a mutation, all queries that were returning the keys returned by the mutation get refetched.

**Comment by JoviDeCroock** [comment]
Well graphcache works a bit different, you can refetch by writing an updater that calls invalidate on the cache but for updates, the cache will update automatically as long as you select the relevant attributes. We have thought about easier ways to transition to graphCache from docCache by means of the populate exchange.

**Comment by JoviDeCroock** [comment]
Closing as stale
